crates/bevy_math/src/bounding/mod.rs

```diff
@@ -0,0 +1,57 @@
+//! This module contains traits and implements for working with bounding shapes
+//!
+//! There are four traits used:
+//! - [`BoundingVolume`] is a generic abstraction for any bounding volume
+//! - [`IntersectsVolume`] abstracts intersection tests against a [`BoundingVolume`]
+//! - [`Bounded2d`]/[`Bounded3d`] are abstractions for shapes to generate [`BoundingVolume`]s
+
+/// A trait that generalizes different bounding volumes.
+/// Bounding volumes are simplified shapes that are used to get simpler ways to check for
+/// overlapping elements or finding intersections.
+///
+/// This trait supports both 2D and 3D bounding shapes.
+pub trait BoundingVolume {
+    /// The position type used for the volume. This should be `Vec2` for 2D and `Vec3` for 3D.
+    type Position: Clone + Copy + PartialEq;
+    /// The type used for the `padded` and `shrunk` methods. For example, an `f32` radius for
+    /// circles and spheres.
+    type Padding;
```

Comment:
This should probably be renamed to something that makes it more obvious it deals with half sizes like the radius of a circle of the half size of a box, rather than the diameter or a full size.